lib/tfpdf/wp-tfpdf.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * To avoid modifying tFPDF directly we introduce WP_tFPDF, 
+ * A thin wrapper around the parent class to utilize WP_Filesystem 
+ * 
+ * Parent's coding standards are preserved deliberately.
+ */
+// phpcs:disable WordPress.WP.AlternativeFunctions.file_system_read_fwrite, WordPress.WP.AlternativeFunctions.file_system_read_fread, WordPress.WP.AlternativeFunctions.file_system_read_fopen, WordPress.WP.AlternativeFunctions.file_system_read_fclose, WordPress.VIP.FileSystemWritesDisallow
+if ( ! defined( 'ABSPATH' ) ) exit; // Exit if accessed directly
+
+require_once ABSPATH . '/wp-admin/includes/file.php';
+
+class WP_tFPDF extends tFPDF {
+	function __construct( $orientation = 'P', $unit = 'mm', $size = 'A4' ) {
+		parent::__construct( $orientation, $unit, $size );
+		$this->init_wp_filesystem();
+	}
+	
+	function init_wp_filesystem() {
+		global $wp_filesystem;
+
+		if ( ! is_a( $wp_filesystem, 'WP_Filesystem_Base' ) ) {
+			$creds = request_filesystem_credentials( site_url() );
+			$fs = wp_filesystem( $creds );
+			return $fs ?: new WP_Error( 'fs-init-error', "Couldn't initialize Filesystem" );
+		}
+		
+		return true;
+	}
+	
+	function Output($name='', $dest='')
+{	
+	// Output PDF to some destination
+	if($this->state<3)
+		$this->Close();
+	$dest = strtoupper($dest);
+	if($dest=='')
+	{
+		if($name=='')
+		{
+			$name = 'doc.pdf';
+			$dest = 'I';
+		}
+		else
+			$dest = 'F';
+	}
+	switch($dest)
```

Comment:
Since the code from here down is identical to the base class's `Output` function except for the one case, could we just call the parent function in any case that is not the one we want to alter the implementation for?

Something like:

```php
switch ( $dest ) {
  case 'F':
    // Custom stuff
    break;

  default:
    parent::Output( $name, $dest );
}
```

Reply:
Great catch, bad copy-pasta is bad :)